util/metric/metric.go

```diff
@@ -205,15 +232,29 @@ func (c *Counter) MarshalJSON() ([]byte, error) {
 	return json.Marshal(c.Counter.Count())
 }
 
+// Desc implements prometheus.Metric.
+func (c *Counter) Desc() *prometheus.Desc {
+	return c.desc
+}
+
+// Write implements prometheus.Metric.
+func (c *Counter) Write(met *prometheus_model.Metric) error {
+	met.Gauge = &prometheus_model.Gauge{Value: proto.Float64(float64(c.Count()))}
```

Comment:
WRT counter vs. gauge: Makes perfect sense.

WRT efficiency: The Prometheus const metric constructors have an overhead of one additional alloc, so in most cases it shouldn't really matter (and I suspect the compiler might optimize it away anyway).

Adding labels or the various buckets of a histogram to a protobuf metric object is really tedious. So `NewConstHistogram` (or `MustNewConstHistogram`, respectively) will really come in handy.

Take into account that the Prometheus histograms are cumulative.